src/testdrive/src/parser.rs

```diff
@@ -534,10 +534,13 @@ impl<'a> Iterator for LineReader<'a> {
             if c == '\n' {
                 self.src_line += 1;
                 if fold_newlines && self.inner.get(i + 1..i + 3) == Some("  ") {
-                    // Chomp the newline and one space. This ensures a SQL query
-                    // that is split over two lines does not become invalid.
+                    // Chomp the two spaces after newline. This ensures a SQL query
+                    // that is split over two lines does not become invalid, and keeping the
+                    // newline ensures that comments don't remove the following lines.
+                    line.push(c);
                     chars.next();
-                    self.pos_map.insert(self.pos + i, (self.src_line, 2));
+                    chars.next();
```

Comment:
It seems like this is controlled by some boolean called `fold_newlines`, so that's now wrong (since we're not folding the newlines anymore).